### PR TITLE
[Trivial] Reorganize `LogicalLocation` code

### PR DESCRIPTION
**Pull request by lroberts36**
## PR Summary

Just move some files around and split up some files to make the code easier to navigate. Also removes a little bit of dead code.
## PR Checklist

- [x] Code passes cpplint
- [x] Code is formatted
- [x] Changes are summarized in CHANGELOG.md
- [x] (@lanl.gov employees) Update copyright on changed files
- [x] #1009 is merged

### CODE REVIEW

**Review by Yurlungur — APPROVED** [review]
Good reorg :+1:

**Comment by pgrete** [comment]
Are you planning to merge this to the forrest branch or later (separately) into develop once forrest is merged?

**Comment by lroberts36** [comment]
@pgrete: I was thinking later after the forest branch is merged. This would just make the diff even bigger I think.

File: src/bvals/neighbor_block.hpp [hunk 1]
```diff
@@ -144,9 +146,6 @@ struct NeighborBlock {
   block_ownership_t ownership;
   RegionSize block_size;
 
-  void SetNeighbor(LogicalLocation inloc, int irank, int ilevel, int igid, int ilid,
-                   int iox1, int iox2, int iox3, NeighborConnect itype, int ibid,
-                   int itargetid, int ifi1 = 0, int ifi2 = 0);
```
Comment (lroberts36) [inline src/bvals/neighbor_block.hpp:149]: This is no longer used anywhere (constructors are used instead), so I removed it.

File: src/mesh/forest/block_ownership.hpp [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+#ifndef MESH_FOREST_BLOCK_OWNERSHIP_HPP_
+#define MESH_FOREST_BLOCK_OWNERSHIP_HPP_
```
Comment (lroberts36) [inline src/mesh/forest/block_ownership.hpp:22]: This previously sat in logical_location.*

File: src/mesh/forest/logical_location.cpp [hunk 3]
```diff
@@ -53,31 +53,6 @@ bool LogicalLocation::Contains(const LogicalLocation &containee) const {
   return (shifted_lx1 == lx1()) && (shifted_lx2 == lx2()) && (shifted_lx3 == lx3());
 }
 
-std::array<int, 3> LogicalLocation::GetOffset(const LogicalLocation &neighbor,
```
Comment (lroberts36) [inline src/mesh/forest/logical_location.cpp:56]: unused

File: src/mesh/forest/logical_location.cpp [hunk 4]
```diff
@@ -358,147 +333,4 @@ std::unordered_set<LogicalLocation> LogicalLocation::GetPossibleNeighborsImpl(
   return unique_locs;
 }
 
-// TODO(LFR): Remove this
-block_ownership_t
-DetermineOwnership(const LogicalLocation &main_block,
```
Comment (lroberts36) [inline src/mesh/forest/logical_location.cpp:363]: just moved for now, next forest PR removes this though.

**Review by pgrete — APPROVED** [review]
LGTM I think this reorg makes sense.

**Pull request merged**